packages/storage/amplify_storage_s3_dart/test/storage_s3_service/transfer/database_html_test.dart

```diff
@@ -0,0 +1,104 @@
+@TestOn('browser')
+import 'package:amplify_storage_s3_dart/src/storage_s3_service/transfer/database/transfer_data.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:test/test.dart';
+
+import '../../test_utils/html_mock.dart';
+
+void main() {
+  group('TransferDatabase for web', () {
+    const testUploadId = 'test-upload-Id';
+    const testObjectKey = 'test-object-Key';
+    final testCreatedAt = DateTime(2022, 1, 1).toIso8601String();
+
+    final testTransferData = TransferData(
+      uploadId: testUploadId,
+      objectKey: testObjectKey,
+      createdAt: testCreatedAt,
+    );
+
+    final testTransferDataJsonString = testTransferData.toJsonString();
+
+    late MockStorage mockStorage;
+    late MockTransferDatabase transferDatabase;
+
+    setUp(() {
+      mockStorage = MockStorage();
+      transferDatabase = MockTransferDatabase(mockStorage);
+    });
+
+    test('insertTransferRecord to local storage', () async {
+      when(
+        () => mockStorage.addAll(any()),
+      ).thenAnswer((_) => {});
+
+      final recordId =
+          await transferDatabase.insertTransferRecord(testTransferData);
+      expect(recordId.startsWith(transferDatabase.prefix), true);
```

Comment:
Could use matchers instead of the raw values.

```suggestion
      expect(recordId.startsWith(transferDatabase.prefix), isTrue);
```